InformationScripting/src/parsing/QueryParser.cpp

```diff
@@ -39,29 +39,45 @@ namespace InformationScripting {
 const QStringList QueryParser::OPEN_SCOPE_SYMBOL{"$<", "\"<", "{<"};
 const QStringList QueryParser::CLOSE_SCOPE_SYMBOL{">$", ">\"", ">}"};
 
-Query* QueryParser::buildQueryFrom(const QString& text, Model::Node* target)
+QList<TopLevelQuery*> QueryParser::buildQueryFrom(const QString& text, Model::Node* target)
 {
 	QueryParser parser;
 	parser.target_ = target;
 	Q_ASSERT(text.size());
-	auto type = parser.typeOf(text);
-	if (Type::Operator == type)
-		return parser.parseOperator(text);
-	else if (Type::Query == type)
-		return parser.parseQuery(text);
-	else if (Type::List == type)
+	// For now assume that we only have yield in operator parts:
```

Comment:
It means the yield operator can only be in an operator scope, i.e. not standalone and not in a list.